LabInfrastructure/LabInfrastructure.csproj

```diff
@@ -0,0 +1,26 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>net4.5.1</TargetFramework>
+
+    <IsPackable>false</IsPackable>
+  </PropertyGroup>
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.8.0" />
+    <PackageReference Include="MSTest.TestAdapter" Version="1.3.2" />
+    <PackageReference Include="MSTest.TestFramework" Version="1.3.2" />
+    <PackageReference Include="NUnit" Version="2.6.4" />
+    <PackageReference Include="Newtonsoft.Json" Version="9.*" />
+    <PackageReference Include="Microsoft.Azure.KeyVault" Version="2.0.6" />
+    <PackageReference Include="Microsoft.Extensions.Configuration" Version="1.1.1" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Binder" Version="1.1.1" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.FileExtensions" Version="1.1.1" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="1.1.1" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <ProjectReference Include="..\adal\src\Microsoft.IdentityModel.Clients.ActiveDirectory\Microsoft.IdentityModel.Clients.ActiveDirectory.csproj" />
```

Comment:
> <ProjectReference Include="..\ada [](start = 3, length = 34)

why this reference?

Reply:
The KeyVaultSecretsProvider class uses some stuff from ADAL (TokenCache and ClientAssertionCertificate)

---
In reply to: [220712127](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1271#discussion_r220712127) [](ancestors = 220712127)

Reply:
can that be extracted? It seems wrong to have that here. perhaps temporarily with links instead of references?

---
In reply to: [220713375](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1271#discussion_r220713375) [](ancestors = 220713375,220712127)

Reply:
ah, we are using ADAL to request a token... I suppose leaving it for now will be fine (we should consider using MSAL instead some time in the future)

---
In reply to: [220717000](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1271#discussion_r220717000) [](ancestors = 220717000,220713375,220712127)

Reply:
still want to switched to link instead of a reference?

---
In reply to: [220717745](https://github.com/AzureAD/azure-activedirectory-library-for-dotnet/pull/1271#discussion_r220717745) [](ancestors = 220717745,220717000,220713375,220712127)